### ISSUE TITLE
E0277 is not detected for `self` parameter in trait methods

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#9446]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4912-222-nightly
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2022.2 Ultimate Edition (IU-222.3345.118)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
The following code contains [E0277](https://doc.rust-lang.org/error-index.html#E0277) error
```rust
trait Foo {
    fn foo(self) {}
}
```

```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
 --> src/main.rs:2:12
  |
2 |     fn foo(self) {}
  |            ^^^^ doesn't have a size known at compile-time
  |
```
but the plugin doesn't detect it in this case although `E0277` is detected in some other cases.

Also, the compiler even suggests some quick fixes how to fix it. The plugin can do it as well in this case.
```
help: consider further restricting `Self`
  |
2 |     fn foo(self) where Self: Sized {}
  |                  +++++++++++++++++
help: function arguments must have a statically known size, borrowed types always have a known size
  |
2 |     fn foo(&self) {}
  |            +
```
